cli/src/depsUpdate.ts

```diff
@@ -0,0 +1,128 @@
+import "zx/globals";
+import subprocess from "node:child_process";
+import {
+  convertToMarkdownTable,
+  createPR,
+  getLatestPackageVersion,
+} from "./util";
+
+export type DepsUpdateConfig = {
+  githubToken: string;
+  githubActor: string;
+};
+
+export async function depsUpdate({
+  githubToken,
+  githubActor,
+}: DepsUpdateConfig) {
+  const packages = getPackages();
+  const toUpdate = new Map<
+    string,
+    { packages: Record<string, string>; latestVersion: string }
+  >();
+  for (const pkg of packages) {
+    for (const dependencies of [
+      pkg.dependencies || {},
+      pkg.devDependencies || {},
+    ]) {
+      for (const [depName, info] of Object.entries(dependencies)) {
+        if (!info.resolved) {
+          // it's a local dependency, like the CLI -- not an NPM package
+          continue;
+        }
+        const existingUpdate = toUpdate.get(depName);
+        if (existingUpdate && existingUpdate.latestVersion !== info.version) {
+          Object.assign(existingUpdate.packages, {
+            [pkg.name]: info.version,
+          });
+          continue;
+        }
+        const latestVersion = await getLatestPackageVersion(depName);
+        if (latestVersion !== info.version) {
+          toUpdate.set(depName, {
+            packages: {
+              [pkg.name]: info.version,
+            },
+            latestVersion,
+          });
+        }
+      }
+    }
+  }
+
+  subprocess.execSync(`git config --global user.name "${githubActor}"`);
+  subprocess.execSync(
+    `git config --global user.email "${githubActor}@users.noreply.github.com"`,
+  );
+  const date = new Date().toISOString().slice(0, 10);
+  const depsToPRs = new Map();
+  for (const [depName, { packages, latestVersion }] of toUpdate) {
+    const head = `syncpack/${depName}-${date}`;
+    const base = "main";
+    const title = `syncpack update ${depName} to ${latestVersion}`;
```

Comment:
Will this approach bypass our syncpack rules for things like `eslint` or different versions of `react` by template?